packages/orama/src/components/index.ts

```diff
@@ -67,9 +75,23 @@ export type VectorIndex = {
   }
 }
 
+export type Tree = {
```

Comment:
Please move these unions into separate type definitions, for example (pseudocode):

```ts
type TreeType =
  | 'avl'
  | 'radix'
  | 'bool'
  | 'flat'

type TTree<T = TreeType, N = unknown> = {
  type: T,
  node: N
}

export type Tree = 
  | TTree<'radix', RadixNode>
  | TTree<'avl',   AVLNode<number, InternalDocumentID[]>>
  | TTree<'bool',  BooleanIndex>
  | TTree<'flat',  FlatTree>
```